### PR TITLE
Customer order, product view

### PR DESCRIPTION
**Pull request by tranthuytan**
sorry, tui gửi trễ

### CODE REVIEW

File: BulkyBookWeb/Controllers/OrderController.cs [hunk 1]
```diff
@@ -11,25 +12,31 @@ namespace BulkyBookWeb.Controllers
...
         public async Task<IActionResult> Index(string? name = "")
-
         {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+                return View();
             IEnumerable<OrderHeader> orders = await unitOfWork.OrderHeaderRepository
-                .GetAll(x => !x.OrderStatus.Equals("Deleted") && x.CustomerName.ToUpper().Contains(name.ToUpper().Trim()));
+                .GetAll(x => User.IsInRole("Admin")
+                            ? !x.OrderStatus.Equals("Deleted") && x.CustomerName.ToUpper().Contains(name.ToUpper().Trim())
```
Comment (ZERO-94) [inline BulkyBookWeb/Controllers/OrderController.cs:30]: why both need !x.OrderStatus.Equals("Deleted") ?

File: BulkyBookWeb/Controllers/ProductController.cs [hunk 2]
```diff
@@ -19,9 +19,12 @@ public ProductController(IUnitOfWork unitOfWork, IWebHostEnvironment webHostEnvi
...
         }
 
-        public async Task<IActionResult> Index()
+        public async Task<IActionResult> Index(string search="")
         {
-            IEnumerable<Product> categories = await unitOfWork.ProductRepository.GetAll(
+            if (!User.IsInRole("Admin"))
+                ViewBag.ShowSearchBar = true;
+            ViewBag.SearchTerm = search;
+            IEnumerable<Product> categories = await unitOfWork.ProductRepository.GetAll(x=> String.IsNullOrEmpty(search) || x.Title.Contains(search.Trim()),
```
Comment (ZERO-94) [inline BulkyBookWeb/Controllers/ProductController.cs]: what not use IsNullOrWhiteSpace ?

File: BulkyBookWeb/Views/Order/Delete.cshtml [hunk 3]
```diff
@@ -60,6 +63,75 @@
...
+				<th>
+					Price
+				</th>
+			</tr>
+		</thead>
+		<tbody>
+			@foreach (var obj in Model.OrderDetails)
+			{
+				<tr>
+					@if (User.IsInRole("Admin"))
```
Comment (ZERO-94) [inline BulkyBookWeb/Views/Order/Delete.cshtml]: no need this, admin doesn't really know about order header

File: BulkyBookWeb/Controllers/ProductController.cs [hunk 4]
```diff
@@ -105,22 +108,22 @@ public async Task<IActionResult> Upsert(ProductViewModel productViewModel, IForm
...
                             }
                             file.CopyTo(fileStream);
                         }
                         product.ImageUrl = Path.Combine("images", "products", fileName + extension);
-                    } 
+                    }
 
                     unitOfWork.ProductRepository.Update(product);
                     var res = await unitOfWork.SaveAsync();
                     TempData["success"] = "Edit successfully!";
```
Comment (ZERO-94) [inline BulkyBookWeb/Controllers/ProductController.cs:122]: make select item for picking status

File: BulkyBookWeb/Views/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -57,6 +57,12 @@
 							<li class="nav-item">
```
Comment (ZERO-94) [inline BulkyBookWeb/Views/Shared/_Layout.cshtml:54]: remove company link

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

File: BulkyBookWeb/Controllers/OrderController.cs (hunk 1 above)
Comment (tranthuytan) [reply BulkyBookWeb/Controllers/OrderController.cs:30]: if else to choose expression, execute query

File: BulkyBookWeb/Views/Order/Delete.cshtml (hunk 3 above)
Comment (tranthuytan) [reply BulkyBookWeb/Views/Order/Delete.cshtml]: remove the orderHeaderId

File: BulkyBookWeb/Controllers/ProductController.cs (hunk 2 above)
Comment (tranthuytan) [reply BulkyBookWeb/Controllers/ProductController.cs]: have changed to IsNullOrWhiteSpace

File: BulkyBookWeb/Controllers/ProductController.cs (hunk 4 above)
Comment (tranthuytan) [reply BulkyBookWeb/Controllers/ProductController.cs:122]: done

File: BulkyBookWeb/Views/Shared/_Layout.cshtml (hunk 5 above)
Comment (tranthuytan) [reply BulkyBookWeb/Views/Shared/_Layout.cshtml:54]: done

**Review by ZERO-94 — APPROVED** [review]

**Pull request merged**